common/app/views/support/Profile.scala

```diff
@@ -96,15 +96,32 @@ object Item640 extends Profile(width = Some(640))
 object Item700 extends Profile(width = Some(700))
 object Video640 extends VideoProfile(width = Some(640), height = Some(360)) // 16:9
 object Video700 extends VideoProfile(width = Some(700), height = Some(394)) // 16:9
-object TwitterImage extends Profile(width = Some(1200))
-object FacebookOpenGraphImage extends Profile(width = Some(1200)) {
 
-  override val heightParam = "h=632"
+object TwitterImage extends Profile(width = Some(1200)) {
+
+  override val heightParam = "h=630"
   val blendModeParam = "bm=normal"
   val blendOffsetParam = "ba=bottom%2Cleft"
   val blendImageParam = "blend64=aHR0cHM6Ly91cGxvYWRzLmd1aW0uY28udWsvMjAxNi8wNi8wNy9vdmVybGF5LWxvZ28tMTIwMC05MF9vcHQucG5n"
   override val fitParam = "fit=crop"
 
+  override def resizeString = {
+    if(TwitterShareImageLogoOverlay.isSwitchedOn) {
+      val params = Seq(widthParam, heightParam, qualityparam, autoParam, sharpParam, fitParam, dprParam, blendModeParam, blendOffsetParam, blendImageParam).filter(_.nonEmpty).mkString("&")
+      s"?$params"
+    } else {
+      super.resizeString
+    }
+  }
+}
+object FacebookOpenGraphImage extends Profile(width = Some(1200)) {
+
+  override val heightParam = "h=630"
+  val blendModeParam = "bm=normal"
+  val blendOffsetParam = "ba=bottom%2Cleft"
+  val blendImageParam = "blend64=aHR0cHM6Ly91cGxvYWRzLmd1aW0uY28udWsvMjAxNi8wNS8yNS9vdmVybGF5LWxvZ28tMTIwMC05MF9vcHQucG5n"
+  override val fitParam = "fit=crop"
+
```

Comment:
`TwitterImage` and `FacebookOpenGraphImage` seems to be duplicate (except the switch).
Why not extracting the switch value then to avoid the duplication:

```
class ShareImage(shouldIncludeLogoOverlay: Boolean) extends Profile(width = Some(1200)) {
  override val heightParam = "h=630"
  val blendModeParam = "bm=normal"
  val blendOffsetParam = "ba=bottom%2Cleft"
  val blendImageParam = "blend64=aHR0cHM6Ly91cGxvYWRzLmd1aW0uY28udWsvMjAxNi8wNi8wNy9vdmVybGF5LWxvZ28tMTIwMC05MF9vcHQucG5n"
  override val fitParam = "fit=crop"

  override def resizeString = {
    if(shouldIncludeLogoOverlay) {
      val params = Seq(widthParam, heightParam, qualityparam, autoParam, sharpParam, fitParam, dprParam, blendModeParam, blendOffsetParam, blendImageParam).filter(_.nonEmpty).mkString("&")
      s"?$params"
    } else {
      super.resizeString
    }
  }
}
object TwitterImage extends ShareImage(TwitterShareImageLogoOverlay.isSwitchedOn)
object FacebookOpenGraphImage extends ShareImage(FacebookShareImageLogoOverlay.isSwitchedOn)
```